jaxrs/src/main/java/org/jboss/as/jaxrs/JaxrsTransformers.java

```diff
@@ -45,12 +56,22 @@ public String getSubsystemName() {
 
     @Override
     public void registerTransformers(SubsystemTransformerRegistration subsystemRegistration) {
-        registerTransformers_2_0_0(subsystemRegistration);
+        if (!isByPassTransformers()) {
+            ChainedTransformationDescriptionBuilder chainedBuilder =
+                    TransformationDescriptionBuilder.Factory.createChainedSubystemInstance(subsystemRegistration.getCurrentSubsystemVersion());
+
+            //Differences between 3.0.0 and 2.0.0
+            ResourceTransformationDescriptionBuilder builder200 =
+                    chainedBuilder.createBuilder(subsystemRegistration.getCurrentSubsystemVersion(), MODEL_VERSION_2_0_0);
+            registerTransformers_2_0_0(builder200);
+
+            //Create chained transformers.
+            chainedBuilder.buildAndRegister(subsystemRegistration, new ModelVersion[] {MODEL_VERSION_2_0_0, MODEL_VERSION_1_0_0});
+        }
```

Comment:
Something feels off here.

AIUI we've actually had a "3.0" (imagine some air quotes) since WF 19, and it looks like we've had a proper 2.0 since some time in 2017. And now you're officially creating a proper 3.0. 

We need transformation for hosts using 1.0 no matter what. We need the missing 3.0 -> 2.0 transformation unless the admin disabled it. If the admin disabled 3.0 -> 2.0 with the system property, then we must do 3.0 -> 1.0.

Maybe rename registerTransformers_2_0_0 to registerTransformersFrom_3_0_0 and then make sure it's invoked no matter what. What differs is the target version provided to chainedBuilder.createBuilder.

BTW I think names like registerTransformersFrom_3_0_0 are better in general. When I review these classes I usually waste time trying to figure out what these version # in these methods represents -- source or target.

Reply:
Okay that makes sense to me. Something felt off to me as well and for some reason I was just not seeing it :) Now that you point it out it's fairly obvious though.

Reply:
Updated. I _think_ I've got it right. For some reason I confuse myself with transformers all the time :)